test/tests/syncRunnerTest.js

```diff
@@ -718,4 +749,20 @@ describe("Zotero.Sync.Runner", function () {
 			assert.isBelow(lastSyncTime, new Date().getTime());
 		})
 	})
+	
+	describe("#getAPIKeyFromCredentials()", function() {
+		it("should return API key when credentials valid", function *() {
+			setResponse('createKey.validCredentials');
```

Comment:
Stubs are fine in the pref tests, but for `syncRunner` we should test the actual data being uploaded — `setResponse` is generally more appropriate for testing GET requests. Look for `server.respond()` in syncEngineTest.js for examples of asserting on the POSTed data.

Should add a test for `#deleteAPIKey()` as well.